src/main/java/duke/dependencies/task/Task.java

```diff
@@ -0,0 +1,241 @@
+    }
+
+    /**
+     * Returns a task representing a todoTask. Task has no date or deadline associated.
+     *
+     * @param task String describing the task.
+     * @return Todo Task.
+     */
+    public static Task createTodo(String task) {
+        return new ToDos(task);
+    }
+
+    /**
+     * Returns a task representing an event. Task will have a date associated with it,
+     * specifying when the event is taking place.
+     *
+     * @param task String describing the task.
+     * @param date Date string, in valid format: "dd/MM/uuuu" or "uuuy-MM-dd".
+     * @return Event Task.
+     */
+    public static Task createEvent(String task, String date) {
+        return new Events(task, date);
+    }
+
+    /**
+     * Returns a task representing a deadline. Deadlines has to be completed by the specified date
+     * De deadline associatione is user referencion.
+     *
+     * @param task String describing the task.
+     * @param date Date string, in valid format: "dd/MM/uuuu" or "uuuy-MM-dd".
+     * @return Deadline Task.
+     */
+    public static Task createDeadline(String task, String date) {
+        return new Deadlines(task, date);
+    }
+
+    /* ----------------------------------------------- Subclasses---------------------------------------------------- */
+
+    private static class EmptyTask extends Task{
+
+        private EmptyTask() {
+            super(null);
+            super.state = null;
+        }
+
+        @Override
+        public String getDateString() {
+            return "";
+        }
+
+        @Override
+        public boolean isItEmpty() {
+            return true;
+        }
+
+        @Override
+        public String toString() {
+            return "There is nothing here to see...";
+        }
+    }
+
+    private static class DoneTask extends Task{ // Should be Misc Task not Done Task
+
+        private DoneTask(String nums) {
+            super(nums);
+            super.state = null;
+        }
+
+        @Override
+        public String getDateString() {
+            return "";
+        }
+
+        @Override
+        public boolean isItEmpty() {
+            return true;
+        }
+
+        @Override
+        public String toString() {
+            return "There is nothing here to see...";
+        }
+    }
+
+    private static class Events extends Task {
+        private String date;
+        private TaskDate tDate;
+
+        private Events(String task, String date) {
+            super(task);
+            this.date = date;
+            this.tDate = new TaskDate(date);
+        }
+
+        @Override
+        public boolean hasADate() {
+            return true;
+        }
+
+        @Override
+        public String getDateString() {
+            return this.tDate.toString();
+        }
+
+        @Override
+        public String toString() {
+            return String.format(
+                    "[Event][%s] %s (at: %s)", super.state == UNFINISHED ? "X"
```

Comment:
Perhaps the last `: "✓",`  line can be in the same line as the previous to improve readability?

Reply:
Agreed, unless the character line limit is reached.